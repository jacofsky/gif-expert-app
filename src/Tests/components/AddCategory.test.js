import React from 'react'
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Test in AddCategory.test.js', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should change input value ', () => {
        const input = wrapper.find('#inputAgregarPalabra')
        const value = 'Hola mundo'

        input.simulate('change', {target: { value }})

        expect(wrapper.find('p').text().trim()).toBe(value)
    })
    
    test('should not post the information on submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}})

        expect(setCategories).not.toHaveBeenCalled()
    })

    test('should call setCategories and clean text case', () => {

        // 1. simular el inputChange
        const input = wrapper.find('#inputAgregarPalabra')
        const value = 'Hola mundo'

        input.simulate('change', {target: {value}})

        // 2. simular el submit
        wrapper.find('form').simulate('submit', {preventDefault(){}})

        // 3. setCategories se debe de haber llamado
        expect(setCategories).toHaveBeenCalled()
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function))

        // 4. el valor del input debe de estar ''
        expect(input.prop('value')).toBe('')
        
    })
    
    
    
})
