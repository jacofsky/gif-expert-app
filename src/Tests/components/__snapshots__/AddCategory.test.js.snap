// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AddCategory.test.js should match the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    id="inputAgregarPalabra"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
